### ISSUE TITLE
[PyTorch] Replace torch.no_grad() with torch.inference_mode()

### ISSUE THREAD

**Issue by mikesvoboda** [mikesvoboda/nemotron-v3-home-security-intelligence#3879]
## Summary

Replace all torch.no_grad() with torch.inference_mode() for 3-10% faster inference.

## Current State

Codebase uses torch.no_grad() throughout ai/ directory.

## Implementation

Simple find-and-replace across 20+ files.

## Files

* ai/rtdetr/model.py
* ai/florence/model.py
* ai/clip/model.py
* backend/services/vitpose_loader.py
* backend/services/xclip_loader.py

## Labels: optimization, ai, pytorch, phase-1
